dev/ese/src/ese/eselibwithtests/CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+# This library is `ese` + tests. See ENABLE_JET_UNIT_TEST being defined below
+# used throughout the code base
+
+# Workaround for CMake's directory scoping; see CMake issue 18399
+add_custom_command(
+    OUTPUT
+        ${GEN_OUTPUT_DIRECTORY}/ese/jetapitable.cxx
+    COMMAND ${CMAKE_COMMAND} -E
+        touch_nocreate ${GEN_OUTPUT_DIRECTORY}/ese/jetapitable.cxx
+    DEPENDS
+        gen_jetapitable
+)
+
+add_custom_command(
+    OUTPUT
+        ${GEN_OUTPUT_DIRECTORY}/ese/eselibwithtests/ese.def
+    COMMAND
+        cl /nologo /c /DLIBNAME=ESElibwithtests /DDEBUGGER_EXTENSION /EP /Tc ${ESE_DEV}/src/ese/esent.def > ${GEN_OUTPUT_DIRECTORY}/ese/eselibwithtests/ese.def
+    DEPENDS
+        ${ESE_DEV}/src/ese/esent.def
+)
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    # Using Visual C++
+
+    # See documentation for C2712
+    string(REPLACE "/EHsc" "/EHs-c-" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+endif()
+
+add_library(eselibwithtests SHARED
+    ${GEN_OUTPUT_DIRECTORY}/ese/eselibwithtests/ese.def
+    ${GEN_OUTPUT_DIRECTORY}/ese/jetapitable.cxx
```

Comment:
the jetapitable.cxx can't be added to ESE_SOURCES like all the other .cxx files because it is generated on fly by build maybe?  just curious.

Reply:
Something like that... I struggled with this part a bunch. There may be more straightforward ways of dealing with generated files, but I haven't tried them yet (e.g., you can mark a file as GENERATED and then CMake treats it in a special way)